src/Agent.Worker/JobExtension.cs

```diff
@@ -162,9 +167,13 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                         }
                     }
 
+                    var taskManager = HostContext.GetService<ITaskManager>();
+
+                    //Check if we need to install old node runners
+                    CheckAndInstallNodeRunner(context, message, taskManager);
```

Comment:
I don't think this addresses my feedback -- My thinking is that installation would occur in the listener as soon as auth to TFS, prior to worker/job starting.